sylvia/tests/ui/method_signature/reply.rs

```diff
@@ -81,4 +81,48 @@ pub mod mismatched_param_arity {
     }
 }
 
+pub mod redundant_params {
+    use super::*;
+
+    pub struct Contract {}
+
+    #[sylvia::contract]
+    #[sv::features(replies)]
+    impl Contract {
+        pub const fn new() -> Self {
+            Self {}
+        }
+
+        #[sv::msg(instantiate)]
+        fn instantiate(&self, _ctx: InstantiateCtx) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[sv::msg(reply, reply_on=success)]
+        fn first_reply(
+            &self,
+            _ctx: ReplyCtx,
+            #[sv::data(opt, raw)] _data: Option<Binary>,
```

Comment:
Let's add few more combinations:
1. What would happen if there is an additional argument before `sv::data`?
2. What happen if we use `#[sv::data(opt, raw)]` on `ctx: ReplyCtx`?
3. What happen if we use `#[sv::payload(raw)]` on `ctx: ReplyCtx`?
4. What if we skip `ctx: ReplyCtx`?
5. Can we use `#[sv::data(raw)]` after `#[sv::payload(raw)]`?

Reply:
6. Missing payload parameter. - This might cause some weird behavior due to deserialization to empty tuple.

Good points.
For 2 and 3 nothing would happen as context types are skipped during parsing.
For 4 as in case of every message type it would fail with Rust compile error that context types couldn't be converted to whatever parameter is defined first. We are unable to check if the second parameter in the method signature is `ReplyCtx` as this would make usage of aliases impossible or very restricted and in the end wrong type could still be passed there.

Added test cases for the rest.